var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -113,7 +116,7 @@ def url_for_version(self, version):
     depends_on('py-numpy', when='+python')
     depends_on('swig', when='+python')
 
-    patch('umfpack_from_suitesparse.patch')
+    patch('umfpack_from_suitesparse.patch', when='@:12.8.1')
```

Comment:
i don't know, but try `when='~@develop'`? @tgamblin @alalazo any better ways?

Otherwise just do `@:9999` for now, because `12.10.1` is out already and most likely need this patch.